### ISSUE TITLE
Wrong File Size conversion in GB

### ISSUE THREAD

**Issue by prasunkhan** [newhinton/disky#22]
If the true file size is 1,073,741,824 B, Disky displays its size as 1.074GB (i.e. simply divide the size by 1000 while converting bytes to KB, KB to MB and MB to GB).

However, the true size in GB is just 1GB, i.e. 1,073,741,824 / (1,024 * 1,024 * 1,024) GB.

**Comment by ParitoshKumarTripathi** [comment]
Yes, this file needs modification - app/src/main/java/de/felixnuesse/disky/extensions/fs.kt

This is the corrected code. (The changes being changing log10(1000) to log10(1024) and 1000.0.pow to 1024.0.pow -

//https://stackoverflow.com/a/5599842
fun Any.readableFileSize(size: Long): String {
    if (size <= 0) return "0"
    val units = arrayOf("B", "KB", "MB", "GB", "TB")
    val digitGroups = (log10(size.toDouble()) / log10(1024.0)).toInt()
    return DecimalFormat("#,##0.#").format(size / 1024.0.pow(digitGroups.toDouble())) + " " + units[digitGroups]
}

**Comment by lord-ne** [comment]
Whether 1GB means 1000^3 or 1024^3 bytes is a matter of debate. Hard drive makers generally choose 1000 since it makes their disk sizesook bigger.

The problem is that the SI prefix "Giga" is supposed to always mean 1000^3. But 1024^3 is generally a much more useful measurement when dealing with disk sizes (because it's a power of 2).

A solution would be to use the SI binary prefixes: KiB, MiB, GiB, TiB, etc. These are unambiguous.

**Comment by newhinton** [comment]
The reason i chose 1000 is because that matches what the default android filemanger shows (on the emulator):

Disky:

<img width="342" height="263" alt="Image" src="https://github.com/user-attachments/assets/d13be60d-20b9-42f9-8c15-fa764f598c8d" />

Android:

<img width="353" height="400" alt="Image" src="https://github.com/user-attachments/assets/249ffa87-6263-4f51-84de-2f4bc385ed46" />


Using 1024 would create a deviation:

<img width="358" height="260" alt="Image" src="https://github.com/user-attachments/assets/d9f350d6-0e5b-4fb1-ad61-0b713b157902" />

I still don't get why anyone would not use powers of 2, but i am pragmatic about it and chose what android does by default to prevent confusion. Did someone have a different experience where disky does not show what your filemanager shows?
